### ISSUE TITLE
Can't register account on Progress Sync on No Touch devices

### ISSUE THREAD

**Issue by whatever4kindle** [koreader/koreader#9093]
Hi,
When I try to register an account for Progress Sync, I can fill the username part, but It wont let me write a password, so I can't register an account and then I cant use this feature.

Tried with Kindle 4 No Touch and Kindle 3 Keyboard, both with the latest Koreader version.

**Comment by dandelions** [comment]
use inner keyboard

**Comment by whatever4kindle** [comment]
> use inner keyboard

What does this mean?

**Comment by pazos** [comment]
https://github.com/koreader/koreader/issues/8024#issuecomment-1157299824

**Comment by Commodore64user** [comment]
just want to say, that this issue has NOT been fixed (or something broke it again). unlike what this says: 

> @pazos It is fixed.
> 
> 1. Screen keyboard hides on `back` pressed.
> 2. Can move focus with array keys
> 3. Screen keyboard shows again on textbox focused

you cannot move focus to the second input box (password usually), it jumps from the first to the cancel button.

**Comment by Frenzie** [comment]
Are you referring to something specific or more generic?

**Comment by Commodore64user** [comment]
Generic, if you have a prompt with two fields (like username and password) one can only add to text to the first one. This ticket mentions a specific case but it is the same for all instances. 

In short, one cannot move the focus to the second field, and type the password.

**Comment by hius07** [comment]
Cannot test but all fields are added
https://github.com/koreader/koreader/blob/4c6919ac2af9a8f0d4f84d80e996871fabc135bd/frontend/ui/widget/multiinputdialog.lua#L137

**Comment by Commodore64user** [comment]
> Cannot test but all fields are added
> 
> https://github.com/koreader/koreader/blob/4c6919ac2af9a8f0d4f84d80e996871fabc135bd/frontend/ui/widget/multiinputdialog.lua#L137

Yet, still doesn't work.

**Comment by hius07** [comment]
Can you add a custom string to the status bar, or a cloud storage, or an opds server?
Can you use MultiInputDialog anywhere?

**Comment by Commodore64user** [comment]
> Can you add a custom string to the status bar, or a cloud storage, or an opds server? Can you use MultiInputDialog anywhere?

No. That is what I mean by, it does not work.

Again, one can edit the first field but not the second one. with the string in the status bar is not problematic as the second field is number of repetitions (what's that all about?), i.e unimportant. But elsewhere you do need both fields as they are usually credentials to log in to things. Currently, you cannot input a password (and I suspect this has been the case for the last two years, at least).

**Comment by Commodore64user** [comment]
just made a break through discovery here, for reference this is what the widget looks like:

```
[     field 1         ]
[     field 2         ]
  [cancel]   [ok]
```

when you first open the widget, the focus is immediately on `filed 1`, so you can edit that field, when you are done you press `back` and the focus is removed (therefore virtual keyboard disappears).

if we now press down (as would expect) to jump to `field 2`, this does not happen, instead we go directly to `[cancel], now here is were the interesting discovery happens, we can bifurcate here.

lets go `up` first, if we do that, the keyboard reappears but the focus is never placed on `field 2` (i.e you can type but no text is actually being added to said field). pressing back here, closes the widget. so dead end.

if however from `[cancel]` we go `down` (so we are in a loop), focus is placed again on `field 1` and we can use the keyboard in said field. Pressing `back` again takes us out of focus but now, _and only now_, pressing `down` puts focus on `field 2` and so can [finally] be edited.

so, not sure what is making it act up like this, but it is sort of broken...

**Comment by hius07** [comment]
If you are going to investigate it further: 1) look how MultiInputDialog populates self.layout, and then 2) how FocusManager iterates over it with designated keys.

**Comment by Commodore64user** [comment]
I am not sure if there is anything in the wrong order, I mean, what's the proper order?

https://github.com/koreader/koreader/blob/7b9c65ee1ed43020473dbcaf321ea85bee34eef5/frontend/ui/widget/multiinputdialog.lua#L112-L137

**Comment by Commodore64user** [comment]
short video, showing focus manager iterating through multi input fields (I am just pressing down multiple times)

https://github.com/user-attachments/assets/3135815e-7862-4633-8de8-ca1ae743aaf6

**Comment by Commodore64user** [comment]
(to be) Fixed (for real now) by #12361
